4788/src/main/include/MagLoader2.h

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include "strategy/StrategySystem.h"
+#include "RobotMap.h"
+
+enum class MagLoaderState {
+  AUTO,
+  SHIFT_IN,
+  MANUAL,
+  FIRE
+};
+
+class MagLoader : public wml::StrategySystem {
+ public:
+  MagLoader(wml::Gearbox &magLoaderGearbox,
+            frc::AnalogInput &startSensor,
+            frc::AnalogInput &indexSensor,
+            frc::AnalogInput &stopSensor);
+
+  void SetMagLoader(const MagLoaderState st, double setpoint) {
+    _magLoaderState = st;
+    _magLoaderSetpoint = setpoint;
+  }
+
+  void UpdateMagLoader(double dt) {
+    double voltage = 0;
+    switch (_magLoaderState) {
+      case MagLoaderState::AUTO:
+        voltage = 12 * _magLoaderSetpoint;
+       break;
+      
+      case MagLoaderState::SHIFT_IN:
+        voltage = 12 * _magLoaderSetpoint;
+       break;
+
+      case MagLoaderState::MANUAL:
+        voltage = 12 * _magLoaderSetpoint;
+       break;
+
+      case MagLoaderState::FIRE:
+        voltage = 12 * _magLoaderSetpoint;
+       break;
+    }
+    _magLoaderGearbox.transmission->SetVoltage(voltage);
+  }
```

Comment:
See if you can implement auto state transitions between AUTO and SHIFT_IN, based on sensors. This is where you'd put the mag auto-loading code from before

Reply:
Transition inside the switchcase?
Not in the strategy file?